src/coreclr/jit/assertionprop.cpp

```diff
@@ -3177,6 +3177,41 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
 
     if (conValTree != nullptr)
     {
+        if (tree->OperIs(GT_LCL_VAR))
+        {
+            bool betterToPropagate = false;
+            if (conValTree->IsVectorZero() || conValTree->IsFloatPositiveZero())
```

Comment:
What about `AllBitsSet` and for `Arm64` those that fit in `fmov`?

Reply:
Good point, will address, just want to see diffs first 🙂

Reply:
I think this is a good direction and will be correct/best for what we have today. As a more meta point. This is probably a case where we actually want to "never propagate constants".

Constant propagation being "better" really depends on the constant being containable as an immediate value. Even for things like `Zero` and `AllBitsSet`, its likely better to hoist them when they can't be contained.

I think this gets back to needing a way to see a given `LCL_VAR` is definitely some `CNS` however, and that's much more complex.

Reply:
If we had a way to see that its a `CNS` then RA could likewise special-case those values as being "cached data" or "cheap to rematerialize" and make an appropriate judgement call on them not requiring spilling and so therefore is a "better" option to use if register selection is problematic.

Reply:
Yes, as @jakobbotsch pointed out there was an attempt to do all of that in LSRA but it also caused some difficulties